community/kernel/src/main/java/org/neo4j/GraphDatabaseBuilder.java

```diff
@@ -0,0 +1,384 @@
+     *
+     * @param bytes the threshold in bytes
+     * @return this builder
+     */
+    public BUILDER withInternalLogRotationThreshold( long bytes )
+    {
+        params.put( GraphDatabaseSettings.store_internal_log_rotation_threshold.name(), String.valueOf( bytes ) );
+        return self();
+    }
+
+    /**
+     * Set the delay since the last rotation before the internal log (messages.log) can be rotated again.
+     *
+     * @param delay the delay in seconds
+     * @return this builder
+     */
+    public BUILDER withInternalLogRotationDelay( int delay )
+    {
+        params.put( GraphDatabaseSettings.store_internal_log_rotation_delay.name(), String.valueOf( delay ) );
+        return self();
+    }
+
+    /**
+     * Set the maximum number of archived internal logs to retain.
+     *
+     * @param count the maximum number of archived internal logs to retain
+     * @return this builder
+     */
+    public BUILDER withInternalLogMaxArchives( int count )
+    {
+        params.put( GraphDatabaseSettings.store_internal_log_max_archives.name(), String.valueOf( count ) );
+        return self();
+    }
+
+    /**
+     * Set the node property keys to index with the legacy auto indexer.
+     *
+     * @param propertyKeys the node property keys to index
+     * @return this builder
+     */
+    public BUILDER withLegacyAutoIndexedNodeProperties( String... propertyKeys )
+    {
+        if ( propertyKeys.length > 0 )
+        {
+            params.put( GraphDatabaseSettings.node_auto_indexing.name(), String.valueOf( true ) );
+            params.put( GraphDatabaseSettings.node_keys_indexable.name(), ArrayUtil.join( propertyKeys, "," ) );
+        }
+        else
+        {
+            params.put( GraphDatabaseSettings.node_auto_indexing.name(), String.valueOf( false ) );
+            params.put( GraphDatabaseSettings.node_keys_indexable.name(), "" );
+        }
+        return self();
+    }
+
+    /**
+     * Set the relationship property keys to index with the legacy auto indexer.
+     *
+     * @param propertyKeys the relationship property keys to index
+     * @return this builder
+     */
+    public BUILDER withLegacyAutoIndexedRelationshipProperties( String... propertyKeys )
+    {
+        if ( propertyKeys.length > 0 )
+        {
+            params.put( GraphDatabaseSettings.relationship_auto_indexing.name(), String.valueOf( true ) );
+            params.put( GraphDatabaseSettings.relationship_keys_indexable.name(), ArrayUtil.join( propertyKeys, "," ) );
+        }
+        else
+        {
+            params.put( GraphDatabaseSettings.relationship_auto_indexing.name(), String.valueOf( false ) );
+            params.put( GraphDatabaseSettings.relationship_keys_indexable.name(), "" );
+        }
+        return self();
+    }
+
+    /**
+     * Disable the background index sampling.
+     *
+     * @return this builder
+     */
+    public BUILDER disableBackgroundIndexSampling()
```

Comment:
To go inline with the other `with...` methods this could actually be called `withoutBackgroundIndexSampling`